<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Introduction" data-toc-modified-id="Introduction-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Introduction</a></span></li><li><span><a href="#Your-task" data-toc-modified-id="Your-task-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Your task</a></span><ul class="toc-item"><li><span><a href="#Load-abalone-dataset-to-dataframe-df" data-toc-modified-id="Load-abalone-dataset-to-dataframe-df-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Load abalone dataset to dataframe df</a></span></li><li><span><a href="#Get-the-summary-of-the-dataset" data-toc-modified-id="Get-the-summary-of-the-dataset-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Get the summary of the dataset</a></span></li><li><span><a href="#how-many-females-and-males-in-this-dataset?" data-toc-modified-id="how-many-females-and-males-in-this-dataset?-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>how many females and males in this dataset?</a></span></li><li><span><a href="#find-the-average-Height-of-Male" data-toc-modified-id="find-the-average-Height-of-Male-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>find the average Height of Male</a></span></li><li><span><a href="#find-the-variance-of-shell-weight-in-the-dataset" data-toc-modified-id="find-the-variance-of-shell-weight-in-the-dataset-2.5"><span class="toc-item-num">2.5&nbsp;&nbsp;</span>find the variance of shell weight in the dataset</a></span></li><li><span><a href="#find-the-row-with-the-heaviest-abolone-in-the-dataset" data-toc-modified-id="find-the-row-with-the-heaviest-abolone-in-the-dataset-2.6"><span class="toc-item-num">2.6&nbsp;&nbsp;</span>find the row with the heaviest abolone in the dataset</a></span></li><li><span><a href="#find-the-variance-of-shell-weight-in-the-dataset" data-toc-modified-id="find-the-variance-of-shell-weight-in-the-dataset-2.7"><span class="toc-item-num">2.7&nbsp;&nbsp;</span>find the variance of shell weight in the dataset</a></span></li></ul></li><li><span><a href="#Solution" data-toc-modified-id="Solution-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Solution</a></span><ul class="toc-item"><li><span><a href="#Load-abalone-dataset-to-dataframe-df" data-toc-modified-id="Load-abalone-dataset-to-dataframe-df-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Load abalone dataset to dataframe df</a></span></li><li><span><a href="#Get-the-summary-of-the-dataset" data-toc-modified-id="Get-the-summary-of-the-dataset-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Get the summary of the dataset</a></span></li><li><span><a href="#how-many-females-and-males-in-this-dataset?" data-toc-modified-id="how-many-females-and-males-in-this-dataset?-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>how many females and males in this dataset?</a></span></li><li><span><a href="#find-the-average-Height-of-Male" data-toc-modified-id="find-the-average-Height-of-Male-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>find the average Height of Male</a></span></li><li><span><a href="#find-the-row-with-the-heaviest-abolone-in-the-dataset" data-toc-modified-id="find-the-row-with-the-heaviest-abolone-in-the-dataset-3.5"><span class="toc-item-num">3.5&nbsp;&nbsp;</span>find the row with the heaviest abolone in the dataset</a></span></li><li><span><a href="#find-the-variance-of-shell-weight-in-the-dataset" data-toc-modified-id="find-the-variance-of-shell-weight-in-the-dataset-3.6"><span class="toc-item-num">3.6&nbsp;&nbsp;</span>find the variance of shell weight in the dataset</a></span></li></ul></li></ul></div>

# An introduction to descriptive statistics

## Introduction

Descriptive statistics are statistics that quantitatively describe or summarise features of a given data set. It can be either a representation of the entire or a sample of a population.

Some measures that are commonly used to describe a dataset are:

- measures of central tendency
- measures of variability or dispersion.

Measures of central tendency include: the mean, median and mode.

Measures of variability include:

- the standard deviation(or variance)
- the minimum and maximum values of the variables 
- kurtosis 
- skewness.



Here are some examples of basic statistics.
* Count
* Mean
* Std
* Min
* Max
* Median
* 25th percentile
* 50th percentile
* 75th percentile

The data exploration step aims to do the followings:
* summarise the main characteristics of the data,
* visualise the data using simple and complex graphs to gain an understanding of the features space and the interactions between them,
* validate the data preparation step.

## Your task

### Load abalone dataset to dataframe df
load `./abalone.csv` to a pandas dataframe. For the dataset information, you could refer to [this aritcle](https://archive.ics.uci.edu/ml/datasets/abalone)

In [34]:
import pandas as pd
abalone_p = './abalone.csv'
abalone = pd.read_csv(
    abalone_p, 
    sep=',', 
    decimal='.', 
    header=0, 
    names=['Sex', 'Length', 'Diameter', 'Height', 
            'Whole weight', 'Shucked weight', 
            'Viscera weight', 'Shell weight', 'Rings'])
#abalone = abalone.apply(pd.to_numeric, errors='coerce')
abalone.head()

Sex  Length  Diameter  Height  Whole weight  Shucked weight  Viscera weight  \
0   M   0.455     0.365   0.095        0.5140          0.2245          0.1010   
1   M   0.350     0.265   0.090        0.2255          0.0995          0.0485   
2   F   0.530     0.420   0.135        0.6770          0.2565          0.1415   
3   M   0.440     0.365   0.125        0.5160          0.2155          0.1140   
4   I   0.330     0.255   0.080        0.2050          0.0895          0.0395   

   Shell weight  Rings  
0         0.150     15  
1         0.070      7  
2         0.210      9  
3         0.155     10  
4         0.055      7

### Get the summary of the dataset
you may use `df.info()` `df.shape` methods

In [35]:
abalone.info()
abalone.shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4177 entries, 0 to 4176
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Sex             4177 non-null   object 
 1   Length          4177 non-null   float64
 2   Diameter        4177 non-null   float64
 3   Height          4177 non-null   float64
 4   Whole weight    4177 non-null   float64
 5   Shucked weight  4177 non-null   float64
 6   Viscera weight  4177 non-null   float64
 7   Shell weight    4177 non-null   float64
 8   Rings           4177 non-null   int64  
dtypes: float64(7), int64(1), object(1)
memory usage: 293.8+ KB


(4177, 9)

### how many females and males in this dataset?
note: `M` stands for male, `F` for female and `I` for infant in the `Sex` column

In [53]:
female = abalone['Sex'] == 'F'
male = abalone['Sex'] == 'M'
infant = abalone['Sex'] == 'I'
infant.value_counts()

False    2835
True     1342
Name: Sex, dtype: int64

#male : 1528
#female:1307
#infant: 1342

### find the average Height of Male

In [54]:
male_height_mean = abalone[male].mean()
male_height_mean

Length             0.561391
Diameter           0.439287
Height             0.151381
Whole weight       0.991459
Shucked weight     0.432946
Viscera weight     0.215545
Shell weight       0.281969
Rings             10.705497
dtype: float64

### find the variance of shell weight in the dataset

In [39]:
shel_weight_variance = abalone['Shell weight'].var()
shel_weight_variance

0.019377383202158645

### find the row with the heaviest abolone in the dataset

In [42]:
max_weight = abalone['Whole weight'].sort_values(ascending= False)
max_weight

891     2.8255
1763    2.7795
1209    2.6570
1051    2.5550
165     2.5500
         ...  
1429    0.0140
237     0.0130
2114    0.0105
238     0.0080
236     0.0020
Name: Whole weight, Length: 4177, dtype: float64

### find the variance of shell weight in the dataset

In [ ]:
# fill in your code here

## Solution

### Load abalone dataset to dataframe df
load `./abalone.csv` to a pandas dataframe. For the dataset information, you could refer to [this aritcle](https://archive.ics.uci.edu/ml/datasets/abalone)

In [44]:
import pandas as pd

In [45]:
df = pd.read_csv('./abalone.csv')

### Get the summary of the dataset
you may use `df.info()` `df.shape` methods

In [46]:
df.describe()


Length     Diameter       Height  Whole weight  Shucked weight  \
count  4177.000000  4177.000000  4177.000000   4177.000000     4177.000000   
mean      0.523992     0.407881     0.139516      0.828742        0.359367   
std       0.120093     0.099240     0.041827      0.490389        0.221963   
min       0.075000     0.055000     0.000000      0.002000        0.001000   
25%       0.450000     0.350000     0.115000      0.441500        0.186000   
50%       0.545000     0.425000     0.140000      0.799500        0.336000   
75%       0.615000     0.480000     0.165000      1.153000        0.502000   
max       0.815000     0.650000     1.130000      2.825500        1.488000   

       Viscera weight  Shell weight        Rings  
count     4177.000000   4177.000000  4177.000000  
mean         0.180594      0.238831     9.933684  
std          0.109614      0.139203     3.224169  
min          0.000500      0.001500     1.000000  
25%          0.093500      0.130000     8.000000  
50%          0.171000      0.234000     9.000000  
75%          0.253000      0.329000    11.000000  
max          0.760000      1.005000    29.000000

In [47]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4177 entries, 0 to 4176
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Sex             4177 non-null   object 
 1   Length          4177 non-null   float64
 2   Diameter        4177 non-null   float64
 3   Height          4177 non-null   float64
 4   Whole weight    4177 non-null   float64
 5   Shucked weight  4177 non-null   float64
 6   Viscera weight  4177 non-null   float64
 7   Shell weight    4177 non-null   float64
 8   Rings           4177 non-null   int64  
dtypes: float64(7), int64(1), object(1)
memory usage: 293.8+ KB


In [48]:
df.shape

(4177, 9)

In [49]:
df.head()

Sex  Length  Diameter  Height  Whole weight  Shucked weight  Viscera weight  \
0   M   0.455     0.365   0.095        0.5140          0.2245          0.1010   
1   M   0.350     0.265   0.090        0.2255          0.0995          0.0485   
2   F   0.530     0.420   0.135        0.6770          0.2565          0.1415   
3   M   0.440     0.365   0.125        0.5160          0.2155          0.1140   
4   I   0.330     0.255   0.080        0.2050          0.0895          0.0395   

   Shell weight  Rings  
0         0.150     15  
1         0.070      7  
2         0.210      9  
3         0.155     10  
4         0.055      7

### how many females and males in this dataset?
note: `M` stands for male, `F` for female and `I` for infant in the `Sex` column

In [50]:
df['Sex'].value_counts()

M    1528
I    1342
F    1307
Name: Sex, dtype: int64

from the output of `.value_counts()` method, we can see there are 1528 males and 1307 females

### find the average Height of Male

In [51]:
df.groupby(['Sex'])['Height'].mean()

Sex
F    0.158011
I    0.107996
M    0.151381
Name: Height, dtype: float64

### find the row with the heaviest abolone in the dataset

In [55]:
# find the index of maximum weight
row_index = df['Whole weight'].idxmax()
row_index

891

In [ ]:
# find the row with the heaviest abolone
df.iloc[row_index, :]

### find the variance of shell weight in the dataset

In [ ]:
df['Shell weight'].var()